core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4334,10 +4334,6 @@ private List<FunctionCall> analyzeWindowFunctions(QuerySpecification node, List<
                     throw semanticException(NOT_SUPPORTED, node, "FILTER is not yet supported for window functions");
                 }
 
-                if (windowFunction.getOrderBy().isPresent()) {
-                    throw semanticException(NOT_SUPPORTED, windowFunction, "Window function with ORDER BY is not supported");
-                }
-
```

Comment:
This check is meant for all window functions including the non-aggregate ones. It should remain and only let window aggregates through.

Reply:
readded in changed shape - but I am not sure if this is accessible. See below

Reply:
Why not accessible? Is it covered in `TestAnalyzer`?

Reply:
That is my understanding